warehouse/models/intermediate/payments/int_payments__refunds_deduped.sql

```diff
@@ -0,0 +1,150 @@
+        SAFE_CAST(NULL AS STRING) AS settlement_status,
+        SAFE_CAST(NULL AS DATE) AS settlement_status_time,
+        SAFE_CAST(NULL AS STRING) AS settlement_reason_code,
+        SAFE_CAST(NULL AS STRING) AS settlement_response_text,
+
+        _line_number,
+        currency_code,
+        instance,
+        extract_filename,
+        ts,
+        littlepay_export_ts,
+        littlepay_export_date,
+        _content_hash,
+        _key,
+        _payments_key,
+        'micropayments' AS source_table
+
+    FROM {{ ref('stg_littlepay__micropayments') }}
+    WHERE type = 'CREDIT'
+        AND _key NOT IN ('043ecc000223a299ce17f6a342b1d240', '3536fb2035bbcf4dcb1f3abf001b5185')
+),
+
+refunds_table_refunds AS (
+    SELECT
+
+        aggregation_id,
+        micropayment_id,
+        participant_id,
+        customer_id,
+        refund_amount,
+        transaction_date,
+        refund_id,
+        settlement_id,
+        retrieval_reference_number,
+        COALESCE(retrieval_reference_number, aggregation_id) AS coalesced_id,
+        transaction_amount,
+        proposed_amount,
+        status,
+        initiator,
+        reason,
+        approval_status,
+        issuer,
+        issuer_comment,
+        created_time,
+        approved_time,
+        settlement_status,
+        settlement_status_time,
+        settlement_reason_code,
+        settlement_response_text,
+        _line_number,
+        currency_code,
+        instance,
+        extract_filename,
+        ts,
+        littlepay_export_ts,
+        littlepay_export_date,
+        _content_hash,
+        _key,
+        _payments_key,
+        'refunds' AS source_table
+
+    FROM {{ ref('stg_littlepay__refunds') }}
+    QUALIFY DENSE_RANK() OVER (PARTITION BY refund_id ORDER BY littlepay_export_ts DESC) = 1
+
+),
+
+refunds_union AS (
+    SELECT *
+    FROM micropayments_table_refunds
+
+    UNION ALL
+
+    SELECT *
+    FROM refunds_table_refunds
+),
+
+int_payments__refunds AS (
+
+    SELECT
+
+        aggregation_id,
+        micropayment_id,
+        participant_id,
+        customer_id,
+        refund_amount,
+        transaction_date,
+        refund_id,
+        settlement_id,
+        retrieval_reference_number,
+        transaction_amount,
+        proposed_amount,
+        status,
+        initiator,
+        reason,
+        approval_status,
+        issuer,
+        issuer_comment,
+        created_time,
+        approved_time,
+        settlement_status,
+        settlement_status_time,
+        settlement_reason_code,
+        settlement_response_text,
+        _line_number,
+        currency_code,
+        instance,
+        extract_filename,
+        ts,
+        littlepay_export_ts,
+        littlepay_export_date,
+        _content_hash,
+        _key,
+        _payments_key,
+        source_table
+
+    FROM refunds_union
+    QUALIFY ROW_NUMBER() OVER (PARTITION BY retrieval_reference_number, refund_amount ORDER BY littlepay_export_ts DESC) = 1
```

Comment:
I think this should be partitioned by `coalesced_id` because otherwise it will be dropping most copies of the rows missing RRN